aloe/result.py

```diff
@@ -0,0 +1,206 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Cucumber-esque outputter for Nose.
+"""
+
+from __future__ import print_function
+from __future__ import unicode_literals
+from __future__ import division
+from __future__ import absolute_import
+# pylint:disable=redefined-builtin, unused-wildcard-import, wildcard-import
+from builtins import *
+# pylint:enable=redefined-builtin, unused-wildcard-import, wildcard-import
+from future import standard_library
+standard_library.install_aliases()
+
+from contextlib import contextmanager
+
+from aloe.registry import (
+    CallbackDecorator,
+    CALLBACK_REGISTRY,
+    PriorityClass,
+)
+from aloe.strings import represent_table
+from blessings import Terminal
+from nose.result import TextTestResult
+
+# A decorator to add callbacks which wrap the steps looser than all the other
+# callbacks.
+# pylint:disable=invalid-name
+outer_around = CallbackDecorator(CALLBACK_REGISTRY, 'around',
+                                 priority_class=PriorityClass.DISPLAY)
+# pylint:enable=invalid-name
+
+
+class StreamWrapper(object):
+    """Used to decorate file-like objects with a handy 'writeln' method"""
+    def __init__(self, term):
+        self.term = term
+
+    def __getattr__(self, attr):
+        if attr in ('term', '__getstate__'):
+            raise AttributeError(attr)
+
+        return getattr(self.term.stream, attr)
+
+    def write(self, arg='', return_=False):
+        """Write to the stream"""
+        self.term.stream.write(arg)
+
+        if return_:
+            self.term.stream.write(self.term.move_up * arg.count('\n'))
+
+    def writeln(self, arg='', return_=False):
+        """Convenience function to write a line to the stream"""
+        self.write(arg + '\n', return_=return_)
+
+
+STREAM_WRAPPER = StreamWrapper(None)
+
+
+@outer_around.each_feature
+@contextmanager
+def feature_wrapper(feature):
+    """Display feature execution."""
+    term = STREAM_WRAPPER.term
+
+    if not term:
+        yield
+        return
+
+    try:
+        if feature.tags:
+            STREAM_WRAPPER.writeln(term.cyan(feature.represent_tags()))
+
+        lines = feature.represented(annotate=False).splitlines()
+        STREAM_WRAPPER.writeln(term.bold_white(lines[0]))
+        STREAM_WRAPPER.writeln()
+        STREAM_WRAPPER.writeln(term.white('\n'.join(lines[1:])))
+        STREAM_WRAPPER.writeln()
+
+        yield
+    finally:
+        pass
```

Comment:
I decided that it's much more readable like this. Else you have an empty yield at the end of a function and it looks very strange.